netlogo-gui/src/main/ide/ShowUsageBox.scala

```diff
@@ -0,0 +1,126 @@
+        usageBox.setVisible(false)
+        dataModel.synchronized(dataModel.setRowCount(0))
+      }
+
+      override def windowGainedFocus(e: WindowEvent): Unit = {}
+    })
+    dataModel.setColumnCount(2)
+    usageTable.setSelectionMode(javax.swing.ListSelectionModel.SINGLE_SELECTION)
+    usageTable.getColumnModel.getColumn(0).setMaxWidth(40)
+    usageTable.getColumnModel.setColumnSelectionAllowed(false)
+    usageTable.getTableHeader().setReorderingAllowed(false)
+    usageTable.setTableHeader(null)
+    usageTable.setRowMargin(3)
+    usageTable.setRowHeight(20)
+//    usageTable.setBackground(Color.WHITE)
+    usageTable.getColumnModel.getColumn(1).setMinWidth(200)
+    usageTable.getColumnModel.getColumn(0).setMinWidth(40)
+    usageTable.setFont(editorArea.getFont)
+
+    usageTable.addMouseListener(new MouseListener(){
+      override def mouseClicked(e: MouseEvent): Unit = {}
+
+      override def mouseExited(e: MouseEvent): Unit = {}
+
+      override def mouseEntered(e: MouseEvent): Unit = {}
+
+      override def mousePressed(e: MouseEvent): Unit = {}
+
+      override def mouseReleased(e: MouseEvent): Unit = {
+        usageBox.setVisible(false)
+        val token = usageTable.getValueAt(usageTable.getSelectedRow, 0).asInstanceOf[Token]
+        dataModel.synchronized(dataModel.setRowCount(0))
+        println(token.start + " " + token)
+        editorArea.setCaretPosition(token.start)
+      }
+    })
+    usageTable.addKeyListener(new KeyListener(){
+      override def keyPressed(e: KeyEvent): Unit = {
+        e.getKeyCode match {
+          case KeyEvent.VK_ESCAPE => usageBox.setVisible(false)
+          case _ => e.setSource(editorArea)
+            editorArea.dispatchEvent(e)
+        }
+      }
+
+      override def keyTyped(e: KeyEvent): Unit = {}
+
+      override def keyReleased(e: KeyEvent): Unit = {}
+    })
+  }
+
+  def showBox(me: MouseEvent, position: Int): Unit = {
+    val tokenOption = getTokenTillPosition(editorArea.getText(), position)
+    for {token <- tokenOption} {
+      val tokens = getUsage(editorArea.getText(), token)
+      dataModel.synchronized(dataModel.setRowCount(0))
+      dataModel.synchronized(
+      for (t <- tokens) {
+        dataModel.addRow(Array[AnyRef](t, editorArea.getLineText(t.start).trim))
+      })
+      if(dataModel.getRowCount != 0) {
+        usageTable.setPreferredScrollableViewportSize(usageTable.getPreferredSize())
+        usageTable.setFillsViewportHeight(true)
+        usageBox.setSize(usageTable.getPreferredSize)
+        usageTable.validate()
+        usageBox.setLocation(me.getLocationOnScreen)
+        usageBox.setVisible(true)
+      }
+    }
+  }
+
+  def getUsage(source: String, token: Token): Seq[Token] = {
+    var list = Seq[Token]()
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    for(t <- iterator) {
+      if(t == token) {
+        list :+= t
+      }
+    }
+    list
+  }
```

Comment:
This is a great question and I actually learned something about scala by figuring out how to answer it! You can find the `equals` implementation of a case class by using  `scala -Xprint:typer`, so I ran `scala -Xprint:typer -e 'case class Token(text: String, value: AnyRef)(val start: Int, val end: Int, val filename: String)'` and got the following:

```
...
          override <synthetic> def equals(x$1: Any): Boolean = Token.this.eq(x$1.asInstanceOf[Object]).||(x$1 match {
  case (_: this.Token) => true
  case _ => false
}.&&({
            <synthetic> val Token$1: this.Token = x$1.asInstanceOf[this.Token];
            Token.this.text.==(Token$1.text).&&(Token.this.value.==(Token$1.value)).&&(Token$1.canEqual(Token.this))
          }))
...
```
Which tells us that the `==` method on token only takes `text`, `tpe`, and `value` into account. `tpe` will (pretty obviously) be the same for both tokens (less obviously, `value` will almost always be the same except in a few corner cases which I don't care to get into). However, `text` is the same only when both strings are the same (it's case-sensitive). In most languages this would be the correct behavior, but NetLogo is case-insensitive, so we need to compare without respect to case.